pallets/pools/src/lib.rs

```diff
@@ -140,27 +217,197 @@ pub struct EpochDetails<BalanceRatio> {
+			EpochSolution::Healthy(solution) => solution.solution.as_slice(),
+			EpochSolution::Unhealthy(solution) => solution.solution.as_slice(),
+		}
+	}
+}
+
+impl<Balance> PartialEq for EpochSolution<Balance>
+where
+	Balance: PartialEq,
+{
+	fn eq(&self, other: &Self) -> bool {
+		match self {
+			EpochSolution::Healthy(s_1) => match other {
+				EpochSolution::Healthy(s_2) => s_1.score == s_2.score,
+				EpochSolution::Unhealthy(_) => false,
+			},
+			EpochSolution::Unhealthy(s_1) => match other {
+				EpochSolution::Healthy(_) => false,
+				EpochSolution::Unhealthy(s_2) => s_1 == s_2,
+			},
+		}
+	}
+}
+
+impl<Balance> PartialOrd for EpochSolution<Balance>
+where
+	Balance: PartialOrd,
+{
+	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+		match self {
+			EpochSolution::Healthy(s_1) => match other {
+				EpochSolution::Healthy(s_2) => {
+					let score_1 = &s_1.score;
+					let score_2 = &s_2.score;
+
+					Some(if score_1 > score_2 {
+						Ordering::Greater
+					} else if score_1 < score_2 {
+						Ordering::Less
+					} else {
+						Ordering::Equal
+					})
+				}
+				EpochSolution::Unhealthy(_) => Some(Ordering::Greater),
+			},
+			EpochSolution::Unhealthy(s_1) => match other {
+				EpochSolution::Healthy(_) => Some(Ordering::Less),
+				EpochSolution::Unhealthy(s_2) => s_1.partial_cmp(s_2),
+			},
+		}
+	}
+}
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct HealthySolution<Balance> {
+	pub solution: Vec<TrancheSolution>,
+	pub score: Balance,
+}
+
+#[derive(Encode, Decode, Clone, Eq, RuntimeDebug, TypeInfo)]
+pub struct UnhealthySolution<Balance> {
+	pub state: Vec<UnhealthyState>,
+	pub solution: Vec<TrancheSolution>,
+	// The risk buffer score per tranche (less junior tranche) for this solution
+	pub risk_buffer_improvement_scores: Option<Vec<Balance>>,
+	// The reserve buffer score for this solution
+	pub reserve_improvement_score: Option<Balance>,
+}
+
+impl<Balance> PartialOrd for UnhealthySolution<Balance>
+where
+	Balance: PartialOrd,
+{
+	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+		// First, we check if any of the risk buffer scores are higher.
+		// A higher risk buffer score for a more senior tranche is more important
+		// than one for a less senior tranche.
+		let senior_to_junior_improvement_scores = self
+			.risk_buffer_improvement_scores
+			.as_ref()
+			.zip(other.risk_buffer_improvement_scores.as_ref());
+		for (s_1, s_2) in senior_to_junior_improvement_scores {
+			if s_1 > s_2 {
+				return Some(Ordering::Greater);
+			} else if s_1 < s_2 {
+				return Some(Ordering::Less);
+			}
+		}
+
+		// If there are no differences in risk buffer scores, we look at the reserve improvement score.
+		if self.reserve_improvement_score > other.reserve_improvement_score {
+			return Some(Ordering::Greater);
+		} else if self.reserve_improvement_score < other.reserve_improvement_score {
+			return Some(Ordering::Less);
+		}
+
+		Some(Ordering::Equal)
+	}
+}
```

Comment:
This is not what we expect from the code. We are zipping it into: `Some(vec[...], vec[...])` in case BOTH solutions are a `MinRiskBufferViolation` and `None` if one or both of them doesn`t. The above will return `Ordering::Greater` in the second case, although, the solution with no `MinRiskBufferViolation` is actually better. 

Hence, I propose the following: 
* Check if both have a `MinRiskBufferViolated` -> Compare the two with simple `<` as rust allows to compare vectors even when wrapped in an `Option` (see reference below).
   * If both scores are equal do the same logic for `MaxReserveViolated`
* If only one of them contains a `MinRiskBufferViolated` the one that doesn`t contain one is greater.

For reference:
```rust
let vec_1 = Some(vec![1,2,3,4,5]);
    let vec_2 = Some(vec![1,3,0,0]);
    let vec_3 = None;
    let vec_4 = None;
    
    assert_eq!(vec_1 > vec_2, false);
    assert_eq!(vec_2 > vec_1, true);
    assert_eq!(vec_1 > vec_3, true);
    assert_eq!(vec_3 == vec_4, true);
    assert_eq!(vec_2 >= vec_4, true);
    
    let val_1 = Some(10u32);
    let val_2 = Some(11u32);
    let val_3 = None;
    let val_4 = None;
    
    assert_eq!(val_1 > val_2, false);
    assert_eq!(val_2 > val_1, true);
    assert_eq!(val_1 > val_3, true);
    assert_eq!(val_3 == val_4, true);
    assert_eq!(val_2 >= val_4, true);
```